fabfile/add.py

```diff
@@ -104,94 +98,93 @@ def get_install_tasks(self):
-        utils.psql("DROP DATABASE IF EXISTS contrib_regression, regression WITH (FORCE);")
+        utils.psql(c, "DROP DATABASE IF EXISTS contrib_regression WITH (FORCE);")
+        utils.psql(c, "DROP DATABASE IF EXISTS regression WITH (FORCE);")
 
         # create result folder if not exists
         utils.mkdir_if_not_exists(config.RESULTS_DIRECTORY)
 
         # load depended extensions
-        self.load_depended_extensions()
+        self.create_depended_extensions(c)
 
-        # run tests with warn_only option to not exit in case of a test failure in the extension,
+        # run tests with warn option to not exit in case of a test failure in the extension,
         # because we want to run regression tests for other extensions too.
-        with settings(warn_only=True):
-            with cd(self.test_path):
-                run(self.test_command)
+        with c.cd(self.test_path):
+            c.run(self.test_command, warn=True)
 
-        # rename regression.diffs, if exists, so that extension's diff file does not conflict with others'        
-        self.rename_regression_diff()
+        # rename regression.diffs, if exists, so that extension's diff file does not override with others'
+        self.rename_regression_diff(c)
 
-    def rename_regression_diff(self):
+    def rename_regression_diff(self, c):
         regression_diffs_path = os.path.join(self.test_path, config.REGRESSION_DIFFS_FILE)
         if os.path.isfile(regression_diffs_path):
-            extension_regression_diff = config.REGRESSION_DIFFS_FILE + '_{}_'.format(self.test_name) + self.extension.name
-            with cd(config.RESULTS_DIRECTORY):
-                run('mv {} {}'.format(regression_diffs_path, extension_regression_diff))
+            extension_regression_diff = config.REGRESSION_DIFFS_FILE + '_{}_{}_'.format(config.PG_VERSION, self.test_name) + self.extension.name
+            with c.cd(config.RESULTS_DIRECTORY):
+                c.run('mv {} {}'.format(regression_diffs_path, extension_regression_diff))
 
 
-class InstallExtensionTask(Task):
+class InstallExtensionTask:
     '''
     A class which has all the boilerplate for building and installing extensions.
     '''
 
     def __init__(self, extension, **kwargs):
-        self.name = extension.name  # the name of the task (fab [xxx])
-        self.__doc__ = extension.doc  # the description which fab --list will list
+        self.name = extension.name
         self.contrib = extension.contrib
         self.repo_url = extension.repo_url
         self.default_git_ref = extension.default_git_ref
         self.preload = extension.preload
+        self.create = extension.create
         self.configure = extension.configure
         self.before_run_hook = kwargs.get('before_run_hook', None)
         self.post_install_hook = kwargs.get('post_install_hook', None)
 
         super(InstallExtensionTask, self).__init__()
```

Comment:
Actually it is designed to be a task because when we `fab add.shard_rebalancer` etc., we should be executing the task for it. But there is no usage in all the project but let this be a todo now in case it can be used by someone.